lib/document.js

```diff
@@ -0,0 +1,1469 @@
+import fetch from 'node-fetch';
+import AWS from 'aws-sdk';
+import { fetchGraphQL, slugify } from './utils';
+const Buffer = require('buffer').Buffer;
+const sizeOf = require('image-size');
+
+export async function processDocumentContents(
```

Comment:
It would be great to write some tests around this function specifically. Definitely the most complicated code in the project.

Reply:
Yes! Definitely need tests on this function.